pkg/cloud-provider/cloudapi/internal/cloud_common.go

```diff
@@ -123,6 +125,20 @@ func (c *cloudCommon) GetCloudAccountByName(namespacedName *types.NamespacedName
 	return accCfg, found
 }
 
+func (c *cloudCommon) GetCloudAccountByAccountId(accountID *string) (CloudAccountInterface, bool) {
+	// accountID is a string representation of namespacedName ie namespace and provider account name joined with a "/".
+	tokens := strings.Split(*accountID, "/")
+	if len(tokens) == 2 {
+		namespacedName := types.NamespacedName{Namespace: tokens[0], Name: tokens[1]}
+		c.mutex.Lock()
+		defer c.mutex.Unlock()
+		accCfg, found := c.accountConfigs[namespacedName]
+		return accCfg, found
+	} else {
+		return nil, false
```

Comment:
add some meaningful error or log, if string tokenize fails

Reply:
Missed while moving from extractNamespacedName function. Updated it.